### ISSUE TITLE
Helm install failing with 'rendered manifests contain a resource that already exists' and connection timed out

### ISSUE THREAD

**Issue by adityagupta1089** [helm/helm#11066]
Output of `helm version`:
```
version.BuildInfo{Version:"v3.9.0", GitCommit:"7ceeda6c585217a19a1131663d8cd1f7d641b2a7", GitTreeState:"clean", GoVersion:"go1.17.5"}
```

Output of `kubectl version`:
```
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.1", GitCommit:"3ddd0f45aa91e2f30c70734b175631bec5b5825a", GitTreeState:"clean", BuildDate:"2022-05-24T12:26:19Z", GoVersion:"go1.18.2", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.4
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.11-gke.900", GitCommit:"20da4c21b3a6b1a56ff6ad5ecb7dee013aaf1b83", GitTreeState:"clean", BuildDate:"2022-03-30T09:37:00Z", GoVersion:"go1.16.15b7", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.24) and server (1.21) exceeds the supported minor version skew of +/-1
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): GKE

I've been trying to install a helm chart using this command:
```
helm --kube-context <cluster_name> install pgs-release kube-prometheus-stack --namespace metric-collection --repo https://prometheus-community.github.io/helm-charts --create-namespace -f metric-collection/helm/prometheus-stack/values.yaml --wait
```
which has been resulting in this error everytime:
```
Error: INSTALLATION FAILED: rendered manifests contain a resource that already exists. Unable to continue with install: could not get information about the resource RoleBinding "pgs-release-grafana" in namespace "metric-collection": Get "https://XX.XX.XX.XXX/apis/rbac.authorization.k8s.io/v1/namespaces/metric-collection/rolebindings/pgs-release-grafana": dial tcp XX.XX.XX.XXX:443: connect: connection timed out
```

Although commands such as `kubectl get pods -A` are working before running this helm command, after `helm install` fails, `kubectl get pods -A` also start failing like this:

## Before
```
> kubectl get pods -A
NAMESPACE     NAME                                                       READY   STATUS    RESTARTS   AGE
kube-system   event-exporter-gke-5479fd58c8-nlp9p                        2/2     Running   0          21d
...
```

## After
```
> kubectl get pods -A
Unable to connect to the server: dial tcp XX.XX.XX.XXX:443: connect: connection timed out
```

**Comment by yxxhero** [comment]
@adityagupta1089 Please check your network.

**Comment by adityagupta1089** [comment]
I don't think there's a problem with that. I had a `ping ...` running in parallel that was working fine.
Strangely it works in local minikube everytime instead of GKE where it fails everytime (I've tried after an hour, two, three and even a day or two). 
I've also tried removing the `wait` debug flag for helm; removing all nodes, services, daemonsets, statefulsets, from the cluster; checking memory/cpu usage (which is well under 10-20%); API usage (under 2%); tried to use `--debug` flag/ `--verbose trace` for more info; removing the specific chart CRDs and reinstalling; using `upgrade --install` instead.
